Работа со строкой запроса

a-rotari opened the pull request:
Added lot's card page

---
:mortar_board: [Работа со строкой запроса](https://up.htmlacademy.ru/php/12/user/1696597/tasks/12)

a-rotari committed:
Module 6 Task 1

Added lot's card page

grabatui commented on lot.php:
```diff
@@ -0,0 +1,56 @@
+    // Prepare to display template
+    $lot_template = include_template('lot.php', ['lot' => $lot]);
+    // Display the template
+    print $lot_template;
```
Php-код не нужно так смещать на ширину таба/4х пробелов. Он должен быть по левому краю файлов.

Это же относится и к `404.php`

a-rotari replied:
Исправил

grabatui commented on templates/404.php:
```diff
@@ -0,0 +1,135 @@
+<html lang="ru">
+<head>
+    <meta charset="UTF-8">
+    <title>DC Ply Mens 2016/2017 Snowboard</title>
```
Думаю тут какой-то другой должен быть тайтл. Хотя бы просто `404`

a-rotari replied:
Исправил

grabatui commented on templates/404.php:
```diff
@@ -0,0 +1,135 @@
+                <li class="nav__item">
+                    <a href="all-lots.html">Разное</a>
+                </li>
+            </ul>
```
Эти типы у тебя уже есть, нужно взять значения из БД. То же самое ниже

a-rotari replied:
Названия теперь беру в БД. Ссылку пока не трогал, так как пока в заданиях этого не было.

grabatui commented on templates/lot.php:
```diff
@@ -0,0 +1,221 @@
+                <li class="nav__item">
+                    <a href="all-lots.html">Разное</a>
+                </li>
+            </ul>
```
Тоже нужно вывести из БД

a-rotari replied:
Названия категорий теперь беру из БД.

grabatui commented on templates/lot.php:
```diff
@@ -0,0 +1,221 @@
+</footer>
+
+</body>
+</html>
```
Все выводимые файлы тебе нужно оборачивать в содержимое `templates/layout.php`. Посмотри как это сделано у тебя в `index.php`

a-rotari replied:
Теперь оборачиваю в layouts.php. Вынес запрос  из БД названий и кодов категорий в отдельную функцию get_categories() в db_functions.php, чтобы этот SQL запрос не писать несколько раз в index.php и в layout.php.

grabatui commented on templates/lot.php:
```diff
@@ -0,0 +1,221 @@
+                        <div class="lot-item__cost-state">
+                            <div class="lot-item__rate">
+                                <span class="lot-item__amount">Текущая цена</span>
+                                <span class="lot-item__cost"><?= $lot['current_price']?></span>
```
Все данные необходимо экранировать перед выводом

a-rotari replied:
Исправил. Теперь всё что выводится в html вроде бы сначала проходит через htmlspecialchars()

a-rotari committed:
Removed indentations after PHP tags

a-rotari committed:
Refactored as per the reviewer's suggestions

Fixed indents after <?php tags in 404.php and in lot.php. Created 'get_categories()' function in db_functions.php to get the categories' names and codes for layout.p
hp. The categories' names in lot.php are now retrieved from the database. All output pages are now wrapped in layout.php providing header and footer. Correct page title now displayed in 404.php.
